### PR TITLE
Async and result code

### PR DESCRIPTION
**Pull request by ddurschlag6river**
Fixes two of my main issues when debugging blueshell.

### CODE REVIEW

File: test/nodes/decorators/Not.test.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ const Action = Behavior.Action;
 const Not = Behavior.decorators.Not;
 
 class EchoAction extends Action<DroneState, string> {
-	onEvent(state: DroneState, event: string) {
-		return event;
+	async onEvent(state: DroneState, event: string) {
+		return event as any;
```
Comment (ddurschlag6river) [inline test/nodes/decorators/Not.test.ts]: Not sure on the intention here. The event type is "string" but we then also return it -- events and result codes being the same seems odd to me.

**Comment by ddurschlag6river** [comment]
We could change the target to es5 if we wanted for compatibility. Sourcemaps should get us back to the modern TS in the debugger, allowing us to breakpoint easily in async methods.

**Comment by jbcpollak** [comment]
@ddurschlag6river - I'm all for the ResultCode change - I wouldn't bother with the async since async operation is deprecated - see `feat/synchronous` - we are probably pretty close to merging that (and making a breaking change to the API) - as soon as the WFM is done with promises, which it should be by now.

**Comment by jbcpollak** [comment]
I recommend closing the PR in favor of https://github.com/6RiverSystems/blueshell/pull/94, I pulled the ResultCode changes into that PR.

**Pull request closed without merging**